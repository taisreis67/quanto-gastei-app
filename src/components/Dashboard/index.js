import React, { Component } from 'react';
import ls from 'local-storage';

class Dashboard extends Component {
    constructor(props) {
        super(props);

        const transactions = ls.get('transactions');
        let totalTransactions = 0;

        if (transactions) {
            const values = transactions.map(transaction => {
                let value = transaction[0]
                        .replace('R$ ', '')
                        .replace('.', '')
                        .replace(',', '.');
    
                return parseFloat(value);
            });
    
            const sum = (total, value) => {
                return total + value;
            }
    
            totalTransactions = values.reduce(sum);
    
            totalTransactions = String(totalTransactions.toFixed(2)).replace('.', ',');
        }

        this.state = {
            totalTransactions,
        };
    }

    render() {
        const {
            totalTransactions,
        } = this.state;

        return (
            <section className="container">
                <div className="row">
                    <div className="col">
                        <div className="card">
                            <div className="card-header">
                                Resumo
                            </div>

                            <div className="card-body">
                                <h5 className="card-title">Saldo total das transações:</h5>
                                <p className="card-text">{`R$ ${totalTransactions}`}</p>
                            </div>
                        </div>
                    </div>
                </div>
            </section>
        );
    }
}

export default Dashboard;