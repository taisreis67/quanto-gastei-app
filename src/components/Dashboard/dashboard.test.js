import React from 'react';
import Dashboard from '../Dashboard';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

it('Deve renderizar o dashboard com o valor total das transações', () => {
    const dashboard = mount(<Dashboard />);

    expect(toJson(dashboard)).toMatchSnapshot();
});