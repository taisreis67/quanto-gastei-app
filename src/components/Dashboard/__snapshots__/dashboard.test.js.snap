// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar o dashboard com o valor total das transações 1`] = `
<Dashboard>
  <section
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div
          className="card"
        >
          <div
            className="card-header"
          >
            Resumo
          </div>
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Saldo total das transações:
            </h5>
            <p
              className="card-text"
            >
              R$ 0
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>
</Dashboard>
`;
